src/frontend/components/property-type/reference/edit.tsx

```diff
@@ -13,67 +13,78 @@ type SelectRecordEnhanced = SelectRecord & {
   record: RecordJSON;
 }
 
-class Edit extends React.Component<CombinedProps> {
-  constructor(props: CombinedProps) {
-    super(props)
-    this.loadOptions = this.loadOptions.bind(this)
-    this.handleChange = this.handleChange.bind(this)
-  }
+const Edit: FC<CombinedProps> = (props) => {
+  const { onChange, property, record, theme } = props
 
-  handleChange(selected: SelectRecordEnhanced): void {
-    const { onChange, property } = this.props
+  const handleChange = (selected: SelectRecordEnhanced): void => {
     if (selected) {
       onChange(property.name, selected.value, selected.record)
     } else {
       onChange(property.name, '')
     }
   }
 
-  async loadOptions(inputValue: string): Promise<Array<SelectRecordEnhanced>> {
-    const { property } = this.props
+  const loadOptions = async (inputValue: string): Promise<SelectRecordEnhanced[]> => {
     const api = new ApiClient()
 
     const records = await api.searchRecords({
       resourceId: property.reference as string,
       query: inputValue,
     })
-    return records.map((record: RecordJSON) => ({
-      value: record.id,
-      label: record.title,
-      record,
+    return records.map((r: RecordJSON) => ({
+      value: r.id,
+      label: r.title,
+      record: r,
     }))
   }
+  const error = record.errors && record.errors[property.name]
 
-  render(): ReactNode {
-    const { property, record, theme } = this.props
-    const error = record.errors && record.errors[property.name]
+  const selectedId = record?.params[property.name] as string | undefined
+  const [loadedRecord, setLoadedRecord] = useState<RecordJSON | undefined>()
+  const [loadingRecord, setLoadingRecord] = useState(0)
+  const selectedValue = record?.populated[property.name] ?? loadedRecord
```

Comment:
question: loadedRecord shouldn't be first here? isn't it more important ten what is in record.populated?

Reply:
No, we only load that data because we don't always have it initially. `populated` field is only there if the property is top-level. Nested properties don't have `populated` so we need to load that data manually. But then when you select a value it will be added to `populated`, so no need to load manually data for it - it will be in `populated`. And the initially loaded data will be stale. That's why it's second, if it wouldn't then the component wouldn't change when selecting a value unless we made that extra request for data we already have.